docs/participation_key_lifecycle.md

```diff
@@ -0,0 +1,162 @@
+files around could easily lead to permission errors.
+
+The current version of **algod** supports a series of endpoints on the Admin API
+to install and manage keys. One of them is a POST for installation:
+
+```
+POST /v2/participation/
+```
+
+For example, install the keys generated above to a node located at **localhost:1234** with the following:
+```
+curl -X POST --data-binary @keys.db -H "Authorization: Bearer admin-token-here" "localhost:1234/v2/participation"
+```
+
+**goal** provides a convenience function for this operation:
+```
+goal account installpartkey --partkey keys.db --delete-input -d /path/to/data-dir
+```
+
+Keys should only be installed on a single node. In the event that multiple
+nodes have the same set of keys, multiple votes could be cast for a single
+account. This could lead to "equivocation" votes, where multiple conflicting
+votes are cast for the same account causing both to be ignored.
+
+## Key Storage
+
+Once installed keys are stored in the **Participation Registry**. This is a
+service that wraps a SQLite file for storage. Once installed, keys are assigned
+an ID. There are additional Admin API endpoints available to manage the
+registry:
+
+```
+DELETE /v2/participation/<participation-id>
+GET /v2/participation/
+GET /v2/participation/<participation-id>
+```
+
+## Key Registration
+
+After installing a key, the account is still considered offline. The network
+does not know about the local keys and will not accept votes until the keys are
+registered with the network. This is done using a special key registration
+transaction.
+
+Like account creation, there are utilities in both **goal** and **algokey**.
+For this document we'll use **algokey**. Because it is designed to use offline
+you need to provide the current round.
+
+It has the following arguments:
+* **network**: One of devnet, betanet, testnet or mainnet.
+* **firstvalid**: The first round where this transaction may be submitted.
+* **keyfile**: The keyfile generated earlier. This is used to source most parameters.
+* **outputFile**: The unsigned keyreg transaction.
+
+Here is an example call that creates **keyreg.txn**, an unsigned keyreg transaction:
+```
+algokey part keyreg --network mainnet --firstvalid 31998000 --keyfile keys.db -o keyreg.txn
+```
+
+The unsigned transaction can also be signed with **algokey**:
+```
+algokey sign -t keyreg.txn -o keyreg.stxn -m “[enter your account’s 25 word private key delimited by spaces]”
+```
+
+Submit the transaction as usual with **goal**.
+
+The node monitors blocks for key registrations and updates the **Participation Registry**.
+
+## Voting
+
+The node asks the registry for keys on each round and uses them as needed. At
+the end of the round it tells the registry to delete them.
+
+## Participation Registry
+
+The participation registry is optimized in several ways to avoid slowing down
+the consensus protocol.
+
+Reads are optimized by caching all keys on startup.
```

Comment:
It would have less than 5000 keys in memory because of the way key dilution works, dilution would be sqrt(20000000), or about 4473.

State proof keys are a little different and are not cached so aggressively.